core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -399,35 +400,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         "Cannot parse suggestion response without type information. Set [" + RestSearchAction.TYPED_KEYS_PARAM
                                 + "] parameter on the request to ensure the type information is added to the response output");
             }
-            Suggestion suggestion = null;
-            Function<XContentParser, Entry> entryParser = null;
-            // the "size" parameter and the SortBy for TermSuggestion cannot be parsed from the response, use default values
-            // TODO investigate if we can use NamedXContentRegistry instead of this switch
-            switch (type) {
-                case Suggestion.NAME:
```

Comment:
I was wrong when I said there are no custom Suggester implementations out there. I found this closed issue https://github.com/elastic/elasticsearch/issues/17100 where we discussed removing the the ability to register custom suggesters, which we didn't do because of plugin projects like this out there: https://github.com/meltwater/es-prefix-suggester. 
When I understand their code right, they only modify the Suggester implementation, which in this case outputs instances of the top level Suggestion type: /blob/master/src/main/java/com/meltwater/elasticsearch/search/suggest/PrefixSuggester.java
One could argue that they need to provide their own parser, but since they use our built-inSuggestion class it might also be reasonable if users assumed this gets properly parsed. Wdyt?